### ISSUE TITLE
[1.16.5] Config resets itself to previous version on restart

### ISSUE THREAD

**Issue by arketec** [GeheimagentNr1/DiscordIntegration#16]
**Versions**
Minecraft Version: 1.16.5
Minecraft Forge Version: 36.2.23
Mod Version: 2.1.0

**Log files**

```
[25.04 22:55:24] [Server] [Thread-1/INFO] [de.ge.di.co.ServerConfig/]: Loading "Discord Integration" Server Config
[25.04 22:55:24] [Server] [Thread-1/INFO] [de.ge.di.co.ServerConfig/]: [active] = true
[25.04 22:55:24] [Server] [Thread-1/INFO] [de.ge.di.co.ServerConfig/]: [bot_token] = <removed>
[25.04 22:55:24] [Server] [Thread-1/INFO] [de.ge.di.co.ServerConfig/]: [channel_id] = <removed>
[25.04 22:55:24] [Server] [Thread-1/INFO] [de.ge.di.co.ServerConfig/]: [command_prefix] = !
[25.04 22:55:24] [Server] [Thread-1/INFO] [de.ge.di.co.ServerConfig/]: [other_bots, transmit_bot_messages] = false
[25.04 22:55:24] [Server] [Thread-1/INFO] [de.ge.di.co.ServerConfig/]: [max_char_count] = -1
[25.04 22:55:24] [Server] [Thread-1/INFO] [de.ge.di.co.ServerConfig/]: [commands] = [SimpleCommentedConfig:{minecraft_command=difficulty, discord_command=difficulty, description=shows the difficulty of the server., enabled=true}, SimpleCommentedConfig:{minecraft_command=discord gamerules, discord_command=gamerules, description=shows the gamerules and their values., enabled=true}, SimpleCommentedConfig:{minecraft_command=discord commands, discord_command=help, description=shows all commands with their description., enabled=true}, SimpleCommentedConfig:{minecraft_command=discord mods, discord_command=mods, description=shows a list of the mods on the server., enabled=true}, SimpleCommentedConfig:{minecraft_command=list, discord_command=online, description=shows how many and which players are on the server., enabled=true}, SimpleCommentedConfig:{minecraft_command=seed, discord_command=seed, description=shows the seed of the active world., enabled=true}, SimpleCommentedConfig:{minecraft_command=time query daytime, discord_command=time, description=shows the current day time on the server., enabled=true}, SimpleCommentedConfig:{minecraft_command=forge tps, discord_command=tps, description=shows the tps statistic of the server and it's dimensions., enabled=true}]
[25.04 22:55:24] [Server] [Thread-1/INFO] [de.ge.di.co.ServerConfig/]: "Discord Integration" Server Config loaded
```

**Describe the bug**
Attempting to add custom commands and every time I restart it loads the last version of the config (the last .bak it seems)

ex config:

```
#Should the Discord integration be active?
active = true
#Token of your Discord bot:
bot_token = "<removed>"
#Channel ID where the bot will be working
#Range: 0 ~ 9223372036854775807
channel_id = <removed>
#Command prefix for Discord commands
command_prefix = "!"
#How long should Discord messages send to Minecraft Chat be at most? If the value is -1, there is no limit to the length.
#Range: -1 ~ 2000
max_char_count = -1

#Messages shown on Discord
[messages]

	#Options for the server crash message
	[messages.server_crashed]
		#Message send to the Discord chat, if the Minecraft server crashed.
		message = "Server crashed"
		#Should a message be sent to the Discord chat, if the server crashed?
		enabled = true

	#Options for the player joined message
	[messages.player_joined]
		#Message send to the Discord chat, if a player joined. (<player name> <message>)
		message = "joined the game."
		#Should a message be sent to the Discord chat, if a player joined?
		enabled = true

	#Options for the server start message
	[messages.server_started]
		#Message send to the Discord chat, if the Minecraft server started.
		message = "Server started"
		#Should a message be sent to the Discord chat, if the server started?
		enabled = true

	#Options for the tamed mob died message
	[messages.tamed_mob_died]
		#Message send to the Discord chat, if a tamed mob died. (<player name> <message>) If left empty, the default Minecraft message is send.
		message = ""
		#Should a message be sent to the Discord chat, if a tamed mob left?
		enabled = false

	#Options for the player left message
	[messages.player_left]
		#Message send to the Discord chat, if a player left the server. (<player name> <message>)
		message = "disconnected."
		#Should a message be sent to the Discord chat, if a player left?
		enabled = true

	#Options for the player got advancement message
	[messages.player_got_advancement]
		#Message send to the Discord chat, if a player got an advancement. (<player name> <message>) **<advancement title>**<new line>*<advancement description>*
		message = "has made the advancement"
		#Should a message be sent to the Discord chat, if a player got an advancement?
		enabled = true

	#Options for the server stop message
	[messages.server_stopped]
		#Message send to the Discord chat, if the Minecraft server stopped.
		message = "Server stopped"
		#Should a message be sent to the Discord chat, if the server stopped?
		enabled = true

	#Options for the player died message
	[messages.player_died]
		#Message send to the Discord chat, if a player died. (<player name> <message>) If left empty, the default Minecraft message is send.
		message = ""
		#Should a message be sent to the Discord chat, if a player died?
		enabled = false

#Options how to deal with other bots
[other_bots]
	#Should messages of other bots be sent to the Minecraft chat?
	transmit_bot_messages = false
	#Command prefixes of other bots. Messages with these prefixes are not sent to the Minecraft chat.
	other_bots_command_prefixes = []

#Command mapping from Discord to Minecraft commands
[[commands]]
	#Minecraft command without prefix ('/')
	minecraft_command = "difficulty"
	#Dicord command without prefix
	discord_command = "difficulty"
	#Description for the help command
	description = "shows the difficulty of the server."
	#Should the command be active?
	enabled = true

[[commands]]
	#Minecraft command without prefix ('/')
	minecraft_command = "discord gamerules"
	#Dicord command without prefix
	discord_command = "gamerules"
	#Description for the help command
	description = "shows the gamerules and their values."
	#Should the command be active?
	enabled = true

[[commands]]
	#Minecraft command without prefix ('/')
	minecraft_command = "discord commands"
	#Dicord command without prefix
	discord_command = "help"
	#Description for the help command
	description = "shows all commands with their description."
	#Should the command be active?
	enabled = true

[[commands]]
	#Minecraft command without prefix ('/')
	minecraft_command = "discord mods"
	#Dicord command without prefix
	discord_command = "mods"
	#Description for the help command
	description = "shows a list of the mods on the server."
	#Should the command be active?
	enabled = true

[[commands]]
	#Minecraft command without prefix ('/')
	minecraft_command = "list"
	#Dicord command without prefix
	discord_command = "online"
	#Description for the help command
	description = "shows how many and which players are on the server."
	#Should the command be active?
	enabled = true

[[commands]]
	#Minecraft command without prefix ('/')
	minecraft_command = "seed"
	#Dicord command without prefix
	discord_command = "seed"
	#Description for the help command
	description = "shows the seed of the active world."
	#Should the command be active?
	enabled = true

[[commands]]
	#Minecraft command without prefix ('/')
	minecraft_command = "time query daytime"
	#Dicord command without prefix
	discord_command = "time"
	#Description for the help command
	description = "shows the current day time on the server."
	#Should the command be active?
	enabled = true

[[commands]]
	#Minecraft command without prefix ('/')
	minecraft_command = "forge tps"
	#Dicord command without prefix
	discord_command = "tps"
	#Description for the help command
	description = "shows the tps statistic of the server and it's dimensions."
	#Should the command be active?
	enabled = true

# New command here
[[commands]]
	#Minecraft command without prefix ('/')
	minecraft_command = "forge entities list"
	#Dicord command without prefix
	discord_command = "entity_list"
	#Description for the help command
	description = "<regex> regex to search entity names"
	#Should everything attached to the Discord command, be attached to the Minecraft command, too?
	use_parameters = true
	#Should the command be active?
	enabled = true
```

**To Reproduce**
Steps to reproduce the behaviour:

1. Start Server
1. wait until config is generate
1. stop server
1. edit config with bot info
1. start server
1. stop server
1. add custom command
1. start server
1. Previous config loads as shown in logs
1. new command doesn't work from discord

**Expected behaviour**
Expected the new config to load

┆Issue is synchronized with this [Trello card](https://trello.com/c/DBO9M6Vn) by [Unito](https://www.unito.io)

**Comment by GeheimagentNr1** [comment]
I had this issue, in different mods. I investigated in different mods, why this occurs, but I could not get a save result.
The only thing I know, it that this issue occurs more often, if the config is change, when the server is running. But more information I don't have at the current state of the investigations.

**Comment by notafroggo** [comment]
has this been solved yet?

**Comment by GeheimagentNr1** [comment]
This issue should hopefully be fixed with version 3.0.0.

Further investigation is not sensible, because the issue is not deterministic and there is no good way to reproduce it.
